woocommerce/woocommerce-ios
# [Mobile Payments] Show Onboarding if not complete when tapping `Collect Payment` on Order Details

Part of: 
- #6607 


### Description


On opening Order Details, this checks the onboarding state for In-Person-Payments.

When the merchant taps `Collect Payment`, it shows onboarding if required.

If onboarding is displayed, when the state becomes `.complete`, the onboarding flow is dismissed and the card payment process started.

There is code added which lays the groundwork for checking the reader connection status first, but at present, only the onboarding state is used.

#### Note on testing

This commit removes the check for a card-present eligible payment gateway account which used to be part of the decision whether to show the `Collect Payment` button on Order Details or not. I did this to make the changes easier to test, and to be closer in line with Android and expected direction of the changes to the `Collect Payment` button.

To restore the previous behaviour, we can revert 3f21f9bb if desired, and we can decide on that at the end of the work towards #6607.


### Testing instructions


Using a store with both Stripe and WCPay plugins installed and activated

1. On the orders tab, tap + and use Create Order
2. Add a product, and tap next to go through to Order Details
3. Tap `Collect Payment`
4. Observe that you see the "Multiple Conflicting Plugins" onboarding screen
5. Use the `Manage Plugins` CTA and disable one of the payment plugins
6. Tap `Done`
7. Observe that you see an onboarding loading screen, which then dismisses and returns to the Order Details
8. Observe that the Collect Payment flow starts.
9. Connect a reader and collect payment.


### Screenshots



https://user-images.githubusercontent.com/2472348/166256431-386246be-d092-40b3-8c58-36d3ad7d2b47.mp4


---
